Replace third-party client PHP

LKaemmerling opened the pull request:
This commit replaces https://github.com/endclothing/prometheus_client_php, as it seems to be without active development anymore, so I forked it and took over the maintaining in https://github.com/lkaemmerling/prometheus_client_php

This client is already widely used (~800 downloads in 1oneweek)

SuperQ commented:
It would be nice to not have a fork of a fork of a fork be the semi-official recommended version. :frowning_face:

LKaemmerling commented:
> It would be nice to not have a fork of a fork of a fork be the semi-official recommended version. :frowning_face: 

Sure, but when both previous owners don't have any interest in maintaining the stuff anymore there is not other solution. I personally commit fully to the maintainership and will maintain the package from now on for ever 😉 I use it in several important projects and already tried to improve it (moving a few big unneeded dependencies out as a sample or I recently bought the book from Brian which gives a few things that are still missing in the client lib and I will add them).
Of course, you will think "Well he can say everything, the maintaining over time matters". True, but as I said I commit to it and to the Prometheus ecosystem in general as I really love it and use it everywhere (even at my home to scrape everything I can get).

brian-brazil commented:
The current listing comes from #1469, so if the current listing is no longer maintained we should check in with the people from there to see what the PHP community is up to in this regard.

Presuming this is indeed unmaintained, I once again do not see an obvious successor given that there appear to be no fewer than 14 single-contributor forks with commits on top of the existing listing, none of which appear to have any community around them.

@NoelDavies @igordrnobrega @bracki

bracki commented:
Can't really tell. I'm not involved anymore, but I can't remember @LKaemmerling having reached out or anything.

bracki commented:
And yes, maybe I should have been more careful when choosing to whom to hand over the project. maybe @rdohms has an opinion.

LKaemmerling commented:
@bracki i wrote to the endoclothing chat three weeks ago and once per mail (as the guy on the support told me) too, I never got an answer from them.

bracki commented:
@LKaemmerling I sent you something on Twitter.

bracki commented:
I'd be completely fine if @LKaemmerling took over, but I'd really would love for @NoelDavies and the other endclothing folks to chime in once.

SuperQ commented:
Maybe it would be good to move this project to a github org (free now) and add additional contributors, consolidate the forks, rather than keep forking things.

LKaemmerling commented:
Good idea. I will create a organization called something like promphp and then move everything(client package + pushgateway client)to this org.

LKaemmerling commented:
FYI i now moved both repositories into an org: https://github.com/PromPHP I already released the specific versions and they are available as of now. I also updated this MR with the new org.

NoelDavies commented:
They are still active, but obviously it's backed from devs at a private company - (I used to work there). So they're having issues during COVID. They're great guys at End.

NoelDavies commented:
> I'd be completely fine if @LKaemmerling took over, but I'd really would love for @NoelDavies and the other endclothing folks to chime in once.

I did mention on all of the PRs and issues, that I'd left END. but see above ^

brian-brazil commented:
Is there at least one other person, preferably from a different company, who also has admin rights on the new organisation? I'd prefer not to have to go through this again in another year's time.

LKaemmerling commented:
@brian-brazil atm not, but of course this is possible if someone volunteers or maybe if you want I can add you (too)

brian-brazil commented:
I've already got enough to maintain thanks.

I'd suggest looking at people who chimed in on https://github.com/endclothing/prometheus_client_php/issues/56, or other recent PRs to that repo.

brian-brazil commented:
Did you manage to find someone to add? It's not a strict requirement, but it'd help to ensure the long-term viability of the repo.

LKaemmerling commented:
@brian-brazil Unfortunately nobody volunteered until now. The only thing i could you would be adding you (or another Prometheus Team Member) to the org as admin. You don't need to maintain/contribute, but this could easily ensure long-term viability.

brian-brazil commented:
I'll merge, but please try and find someone. Thanks!

rdohms commented:
@LKaemmerling I'll reach out to my network and look inside SurveyMonkey if we have someone to get involved. We depend on the original library so if we move to the new ones we will indeed be interested in making this stick.

@NoelDavies would you or one of your previous colleagues be willing to get involved? Or to deprecate the package in favor of this one to have a clean slate?

Let's also trace all this on composer and ensure we are using the `abandoned` tag to point to new packages so we can move the stragglers.

LKaemmerling commented:
> I'll merge, but please try and find someone. Thanks!

@brian-brazil: Thank you, yeah will find someone :)

> @LKaemmerling I'll reach out to my network and look inside SurveyMonkey if we have someone to get involved. We depend on the original library so if we move to the new ones we will indeed be interested in making this stick.
> 
> @NoelDavies would you or one of your previous colleagues be willing to get involved? Or to deprecate the package in favor of this one to have a clean slate?
> 
> Let's also trace all this on composer and ensure we are using the `abandoned` tag to point to new packages so we can move the stragglers.

@rdohms: Thank you too! It would be great if you (or someone else from SurveyMonkey or your network) would join, this way we won't depend on a single company maintaining the lib like with jimdoo or endoclothing.  The only "requirement" i see is that the people who join needs to know about Prometheus. I already got a few offers for help, but they all didn't use Prometheus (or even the library itself), so this wouldn't be a big benefit for the lib then.

Just for the record: The adoption of the package is already quite good. 
<img width="1613" alt="Bildschirmfoto 2020-10-15 um 06 51 06" src="https://user-images.githubusercontent.com/4281581/96078605-15a4f880-0eb3-11eb-9be9-b9f3c4f57c72.png">
(Packagist Downloads, of course, exported/scraped with the library ;))

bracki commented:
@LKaemmerling just add me to the `promphp` org as well. I won't really be active, but can be there as a backup if it helps.
